gcapi/client.py

```diff
@@ -640,24 +644,34 @@ def upload_cases(  # noqa: C901
 
         return raw_image_upload_session
 
-    def run_external_job(  # noqa: C901
-        self, *, algorithm: str, inputs: dict[str, Any]
+    def run_external_job(
+        self,
+        *,
+        algorithm: Union[str, gcapi.models.Algorithm],
+        inputs: SocketValueSetDescription,
     ):
         """
         Starts an algorithm job with the provided inputs.
+
         You will need to provide the slug of the algorithm. You can find this in the
         url of the algorithm that you want to use. For instance, if you want to use
         the algorithm at
             https://grand-challenge.org/algorithms/corads-ai/
         the slug for this algorithm is "corads-ai".
-        For each input interface defined on the algorithm you need to provide a
-        key-value pair (unless the interface has a default value),
-        the key being the slug of the interface, the value being the
-        value for the interface. You can get the interfaces of an algorithm by calling
+
+        For each input socket defined on the algorithm you need to provide a
+        key-value pair (unless the socket has a default value),
```

Comment:
```suggestion
        key-value pair,
```

Reply:
We no longer allow omitting default values

Reply:
I believe the create strategy checks for this correctly, but probably good to double check!

Reply:
Double checked, and indeed!